src/bin/driver.rs

```diff
@@ -309,46 +258,36 @@ mod app {
                     [digital_inputs.0.is_high(), digital_inputs.1.is_high()];
                 telemetry.digital_inputs = digital_inputs;
 
-                let hold = settings.force_hold
-                    || (digital_inputs[1] && settings.allow_hold);
+                let hold = settings.output.low_noise.force_hold
+                    || (digital_inputs[1]
+                        && settings.output.low_noise.allow_hold);
 
                 (adc0, adc1, dac0, dac1).lock(|adc0, adc1, dac0, dac1| {
-                    let adc_samples = [adc0, adc1];
-                    let dac_samples = [dac0, dac1];
-
                     // Preserve instruction and data ordering w.r.t. DMA flag access.
                     fence(Ordering::SeqCst);
 
-                    for channel in 0..adc_samples.len() {
-                        adc_samples[channel]
-                            .iter()
-                            .zip(dac_samples[channel].iter_mut())
-                            .map(|(ai, di)| {
-                                let x = f32::from(*ai); // get adc sample in volt
-                                let iir = if output[channel].is_enabled() {
-                                    settings.iir_ch[channel]
-                                } else {
-                                    *output[channel].iir()
-                                };
-                                let y = iir.update(
-                                    &mut iir_state[channel],
-                                    x,
-                                    hold,
-                                );
-
-                                // Convert to DAC code. Output 1V/1A Driver output current. Bounds must be ensured by filter limits!
-                                *di = DacCode::try_from(y).unwrap().0;
-
-                                write_dac_spi::spawn(
-                                    channel.try_into().unwrap(),
-                                    y,
-                                )
-                                .unwrap();
-                            })
-                            .last();
-                    }
+                    // Perform processing for  ADC channel 0 --> Driver channel 0 (LowNoise)
+                    let x = f32::from(adc0[0]); // get adc sample in volt
```

Comment:
Probably goo didea to apply the AFE gain here (you have the settings) to avoid confusion.